src/libutil/memory-source-accessor.cc

```diff
@@ -135,11 +135,11 @@ SourcePath MemorySourceAccessor::addFile(CanonPath path, std::string && contents
 
     auto * f = open(path, File{File::Regular{}});
     if (!f)
-        throw Error("file '%s' cannot be made because some parent file is not a directory", path);
+        throw Error("file '%s' cannot be created because some parent file is not a directory", showPath(path));
     if (auto * r = std::get_if<File::Regular>(&f->raw))
         r->contents = std::move(contents);
     else
-        throw Error("file '%s' is not a regular file", path);
+        throw NotARegularFile("file '%s' is not a regular file", showPath(path));
```

Comment:
should this be "cannot be created because it already exists and is not a regular file" or something like that?